### PR TITLE
Update K8s cluster deployment

### LINKED ISSUE #326: [Feature] Update K8s cluster deployment
### Search before asking

- [X] I searched the [issues](https://github.com/IBM/data-prep-lab/issues) and found no similar issues.

### Component

Other

### Feature

Currently K8s cluster deployment is done from  `kind` - the root project subdirectory. 

- It contains only Makefile, Readme and dependencies files. All installation scripts are in ROOT_DIR/scripts/kind
- It can be used to deploy dependencies (KFP, KubeRay) on an external K8s cluster, so its name is not always true.  
- Its Makefile contains KFP version settings, despite that it is defined now in .make.defaults. 

The suggestion is to move the content of the `kind` directory into `scripts/kind` and possibly rename the last dir to something like `scripts/k8s_setup`

Add a Make rule that describes and prints the relevant environment variables, similar to `extra-help` in `.make.transformers`   

Because the `scripts` directory doesn't have the Makefile, the Kubernetes cluster setup/cleanup will be separated from the root make directives. 
 
### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by revit13**
## Why are these changes needed?

/Closes #326 

## Related issue number (if any).

#326

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -90,7 +90,7 @@ jobs:
             - name: Test KFP libs (shared and v1) and run a workflow
               timeout-minutes: 120
               run: |
-                  source kind/requirements.env
+                  source scripts/k8s-setup/requirements.env
```
Comment (roytman) [inline .github/workflows/test.yml]: should we define `scripts/k8s-setup` as a variable?

File: .github/workflows/workflow-manual-run.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ jobs:
             - name: Test KFP worflow run
               timeout-minutes: 120
               run: |
-                  source kind/requirements.env
+                  source scripts/k8s-setup/requirements.env
```
Comment (roytman) [inline .github/workflows/workflow-manual-run.yml]: here we mention `scripts/k8s-setup` only twice, but might be better to have a variable anyway

File: .make.versions [hunk 3]
```diff
@@ -77,8 +77,8 @@ KFP_DOCKER_VERSION_v2=$(DPK_VERSION)
 ################## ################## ################## ################## ################## ##################
 # Begin versions that the repo depends on. 
 
-KFP_v2=2.7.0
-KFP_v1=1.8.22
+KFP_v2=2.2.0
```
Comment (roytman) [inline .make.versions:80]: did you have any issues with kfp 2.7.0 ?

File: kfp/doc/setup.md [hunk 4]
```diff
@@ -78,7 +78,7 @@ Now, you can create a Kind cluster with all required software installed using th
 ```shell
  make setup
 ```
-from this main package directory or from the `kind` directory.
+from this main package directory or from the `scripts/k8s-setup` directory.
```
Comment (roytman) [inline kfp/doc/setup.md]: it should be executed from the `scripts/k8s-setup` directory, and not from the root dir.
The comment is relevant to the other commands below.

File: scripts/k8s-setup/Makefile [hunk 5]
```diff
@@ -1,30 +1,42 @@
...
 KIND_CLUSTER_NAME ?= dataprep
 
+# The following taken from https://stackoverflow.com/a/65243296/45375
+# Lists all targets and optional help text found in the target.
+# :: rule means we first run the help from the main make.defaults.
+extra-help:	
+	@# Help: Shows additional make macros that can be overridden/used. 
+	@echo ""
+	@echo Overridable macro values include the following:
+	@echo EXTERNAL_CLUSTER - If 0 create a KinD cluster.  EXTERNAL_CLUSTER=$(EXTERNAL_CLUSTER)
```
Comment (roytman) [inline scripts/k8s-setup/Makefile]: if not `1` - create a KinD cluster

File: scripts/k8s-setup/Makefile [hunk 6]
```diff
@@ -1,30 +1,42 @@
...
+	@echo DEPLOY_KUBEFLOW - If 1 deploy kubeflow pipelines on the cluster. DEPLOY_KUBEFLOW=$(DEPLOY_KUBEFLOW)
+	@echo USE_KFP_MINIO - If 1 and DEPLOY_KUBEFLOW is also 1 then deploy MinIO on the cluster including ingress to access it.  USE_KFP_MINIO=$(USE_KFP_MINIO)
```
Comment (roytman) [inline scripts/k8s-setup/Makefile]: then use the KFP Minio server for S3 tests

File: scripts/k8s-setup/Makefile [hunk 7]
```diff
@@ -1,30 +1,42 @@
...
+	@echo POPULATE_TEST_DATA - If 1 populate test data to MinIO.  POPULATE_TEST_DATA=$(POPULATE_TEST_DATA)
```
Comment (roytman) [inline scripts/k8s-setup/Makefile]: we can use an external Minio Server $MINIO_SERVER, but maybe better to define an S3 server and provide its username and password, see "populate_minio.sh"

File: .make.versions (hunk 3 above)
Comment (revit13) [reply .make.versions:80]: [kfp/pipelines](https://github.com/kubeflow/pipelines) repo does not have tag 2.7.0

File: .github/workflows/test.yml (hunk 1 above)
Comment (revit13) [reply .github/workflows/test.yml]: Fixed. Thanks

File: .github/workflows/workflow-manual-run.yml (hunk 2 above)
Comment (revit13) [reply .github/workflows/workflow-manual-run.yml]: Fixed. Thanks

File: kfp/doc/setup.md (hunk 4 above)
Comment (revit13) [reply kfp/doc/setup.md]: Fixed. Thanks

File: scripts/k8s-setup/Makefile (hunk 5 above)
Comment (revit13) [reply scripts/k8s-setup/Makefile]: Fixed. Thanks

File: scripts/k8s-setup/Makefile (hunk 6 above)
Comment (revit13) [reply scripts/k8s-setup/Makefile]: Fixed. Thanks

File: kfp/README.md [hunk 8]
```diff
@@ -4,20 +4,23 @@
...
-| code/programming language_annotator | [proglang_select_wf.py](../transforms/code/proglang_select/kfp_ray/v1/proglang_select_wf.py) |
-| universal/doc_id                    |                  [doc_id_wf.py](../transforms/universal/doc_id/kfp_ray/v1/doc_id_wf.py)                   |
-| universal/ededup                    |                  [ededup_wf.py](../transforms/universal/ededup/kfp_ray/v1/ededup_wf.py)                   |
-| universal/fdedup                    |                  [fdedup_wf.py](../transforms/universal/fdedup/kfp_ray/v1/fdedup_wf.py)                   |
-| universal/filtering                 |              [filter_wf.py](../transforms/universal/filter/kfp_ray/v1/filter_wf.py)              |
-| universal/noop                      |                     [noop_wf.py](../transforms/universal/noop/kfp_ray/v1/noop_wf.py)                      |
-| universal/tokenization              |         [tokenization_wf.py](../transforms/universal/tokenization/kfp_ray/v1/tokenization_wf.py)          |
+| language/lang_id                  |                  [lang_id_wf.py](../transforms/language/lang_id/kfp_ray/lang_id_wf.py)                   |
```
Comment (daw3rd) [inline kfp/README.md:7]: This large list will be hard to keep in sync. Do we really want it here or can we link to the transforms README maybe?  But I won't hold up this PR over that :).

File: transforms/language/lang_id/ray/Makefile [hunk 9]
```diff
@@ -34,7 +34,7 @@ setup:: .transforms.setup
 
 # set the version of python transform that this depends on.
 set-versions: 
-	$(MAKE) TRANSFORM_PYTHON_VERSION=${LANG_ID_PYTHON_VERSION} TOML_VERSION=$(LANG_ID_RAY_VERSION) .transforms.set-versions
+	$(MAKE) TOML_VERSION=$(DOCKER_IMAGE_VERSION) .defaults.update-toml
```
Comment (daw3rd) [inline transforms/language/lang_id/ray/Makefile]: I think this should be using `.transforms.set-versions`, although I don't believe that target uses TOML_VERSION.

**Review by daw3rd — DISMISSED** [review]

File: transforms/language/lang_id/ray/Makefile (hunk 9 above)
Comment (revit13) [reply transforms/language/lang_id/ray/Makefile]: Fixed. thanks

**Review by daw3rd — COMMENTED** [review]
LGTM

**Review by roytman — APPROVED** [review]
LGTM

**Pull request merged**